sync-core/src/main/java/com/cloudant/sync/datastore/RevisionHistoryHelper.java

```diff
@@ -195,7 +195,8 @@ public static MultipartAttachmentWriter createMultipartWriter(BasicDocumentRevis
                         mpw = new MultipartAttachmentWriter();
                         mpw.setBody(revision);
                     }
-                    mpw.addAttachment(att);
+                    mpw.addAttachment(savedAtt,
+                            savedAtt.encoding == Attachment.Encoding.Plain ? savedAtt.length : savedAtt.encodedLength);
```

Comment:
I'm glad you mentioned this because I had similar thoughts. I was trying to keep as closely as possible to the couch metadata model but this nasty pattern arises. As suggested I think I'll add a method to SavedAttachment called "onDiskLength" because that's as literal and clear as you can get.